main.go

```diff
@@ -91,13 +108,31 @@ func main() {
 
 	validationConfig := initValidationConfig()
 
-	err = (&controllers.RuleReconciler{
+	if sideCarMode {
+		operator = &controllers.FilesOperator{
+			Log:           ctrl.Log.WithName("controllers").WithName("Rule"),
+			RulesFilePath: rulesFilePath,
+		}
+	} else {
+		operator = &controllers.ConfigMapOperator{
+			Client: mgr.GetClient(),
+			Log:    ctrl.Log.WithName("controllers").WithName("Rule"),
+			DefaultConfigMap: types.NamespacedName{
+				Name:      rulesConfigmapName,
+				Namespace: rulesConfigmapNamespace,
+			},
+			RulesFileName: rulesFileName,
+		}
+	}
+
+	ruleReconciler := &controllers.RuleReconciler{
 		Client:           mgr.GetClient(),
 		Log:              ctrl.Log.WithName("controllers").WithName("Rule"),
-		RuleConfigmap:    types.NamespacedName{Name: rulesConfigmapName, Namespace: rulesConfigmapNamespace},
 		ValidationConfig: validationConfig,
-		RulesFileName:    rulesFileName,
-	}).SetupWithManager(mgr)
+		OperatorMode:     operator,
+	}
+
+	err = ruleReconciler.SetupWithManager(mgr)
 	if err != nil {
 		setupLog.Error(err, "unable to create controller", "controller", "Rule")
```

Comment:
Does `setupLog` not support a fatal command?

Reply:
From what I have found not, no it does not :<